### DISCUSSION TITLE
AntDesign styles being duplicated during webpack code split

### DISCUSSION THREAD

**Discussion by douglasjunior** [ant-design/ant-design#29818] [General]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
I need a recommendation: how to configure the code split of the webpack to AntDesign styles are not duplicated. 

My project was created using `create-react-app` and we use `craco` to configure `less-loader` and `babel-plugin-import`.  Also, we are using `react-router`, where all routes are imported with `lazy (() => import (...))`. 

![image](https://user-images.githubusercontent.com/1512341/111696331-f5d38100-8812-11eb-9a43-6f3a61680ac3.png)

By default, I expect that all routes to be separated into chunks and that common components to be extracted in reusable chunks, but that is not what is happening.

For example, the `ant-tag` style class can be found in more than one chunk, causing them to be interpreted more than once by the browser. 

![image](https://user-images.githubusercontent.com/1512341/111696617-48ad3880-8813-11eb-8a20-54b2fa18eb04.png)

![image](https://user-images.githubusercontent.com/1512341/111696825-8611c600-8813-11eb-83f9-3ded64c63e82.png)

### What does the proposed API look like?
The solutions I found were:

1. Configure a `splitChunks.cacheGroup` to unify all AntDesign styles in one file, but this causes the code split to lose its usefulness. And this also resulted in the generation of a JavaScript chunk, with the same name as the CSS chunk, containing only empty functions (perhaps due to the use of LESS).

```js
           splitChunks: {
                chunks: 'all',
                name: false,
                cacheGroups: {
                    antdStyles: {
                        test: /[\\/]node_modules[\\/]antd[\\/].*\.(s?css|less)$/,
                        name: 'antdStyles',
                        chunks: 'all',
                        reuseExistingChunk: true,
                    },
                },
           },
```

![image](https://user-images.githubusercontent.com/1512341/111697450-58794c80-8814-11eb-9ce9-c2b861defd75.png)

2. Increase the `splitChunks.maxAsyncRequests` size from 30 to `Number.MAX_SAFE_INTEGER`, but I don't know if this is a good idea. I think that this could end up becoming a performance problem. 

```js
           splitChunks: {
                chunks: 'all',
                name: false,
                maxAsyncRequests: Number.MAX_SAFE_INTEGER,
           },
```

So, I come to ask for recommendations on how to configure the webpack, to only chunks without duplication be created, without compromising performance.

**Comment by afc163** [comment]
Try to drop usage of babel-plugin-import.

**Reply by douglasjunior** [reply]
The duplicity is gone, but I end up importing all unused components without babel-plugin-import. 

And that doesn't prevent duplication from happening in my own components, or even in other libraries. 😢

I would like to know how people are solving this in their projects.

**Comment by zombieJ** [comment]
Since this not a bug or feature request. Move to discussion instead.

**Comment by joseooortega** [comment]
Did you find any solution for this? 🤞 🙏
